src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -215,11 +215,17 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
             return null;
         }
 
-        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out IntPtr dstIntPtr)
+        private static Exception ConvertPointerIfPossible(object srcObject, EETypePtr dstEEType, CheckArgumentSemantics semantics, out object dstPtr)
         {
-            if (srcObject is IntPtr srcIntPtr)
+            if (srcObject is IntPtr)
             {
-                dstIntPtr = srcIntPtr;
+                dstPtr = srcObject;
+                return null;
+            }
+
+            if (srcObject is UIntPtr)
+            {
+                dstPtr = srcUIntPtr;
```

Comment:
Where is `srcUIntPtr` defined? Should this be `dstPtr = srcObject;`? If so I think it would be better just to combine them.

```csharp
            if (srcObject is IntPtr || srcObject is UIntPtr)
            {
                dstPtr = srcObject;
                return null;
            }
```